plugins/woocommerce/src/Blocks/BlockTypes/SingleProduct.php

```diff
@@ -156,6 +161,43 @@ protected function replace_post_for_single_product_inner_block( $block, &$contex
 		}
 	}
 
+	/**
+	 * Render the Single Product block.
+	 *
+	 * @param array    $attributes Block attributes.
+	 * @param string   $content Block content.
+	 * @param WP_Block $block Block instance.
+	 *
+	 * @return string Rendered block type output.
+	 */
+	protected function render( $attributes, $content, $block ) {
+		$product = wc_get_product( $block->context['postId'] );
+
+		if ( ! $product instanceof \WC_Product ) {
+			return '';
+		}
+
+		if ( ! $product->is_type( 'variable' ) ) {
+			return parent::render( $attributes, $content, $block );
+		}
+
+		$interactivity_context = array(
+			'originalProductData' => ProductDataUtils::get_product_data( $product ),
+			'productData'         => array(),
+		);
+
+		$html = new \WP_HTML_Tag_Processor( $content );
+
+		if ( $html->next_tag( array( 'tag_name' => 'div' ) ) ) {
+			$html->set_attribute( 'data-wp-interactive', 'woocommerce/product-data' );
```

Comment:
I don't think we should set the namespace here to `woocommerc/product-data`, it's should be `woocommerce/single-product`.

Reply:
It's not a hard rule, but I think we should distinguish between global store like `woocommerce` or `woocommerce/product-data` that only deal with data and local store attached to specific block that can have actions/callbacks. Imagine if later on, we want to add some actions to the Single Product block, what should we do?

Reply:
Yes, it makes sense! I changed it in c6883be5293c620ce72fc50f810f258c700e35b2.